src/modules/payments/services/payments.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Payment } from '../entities/payment.entity';
+import { Injectable } from '@nestjs/common';
+import { PaymentRecord } from '../types/payment-record.type';
+
+@Injectable()
+export class PaymentsService {
+  constructor(
+    @InjectRepository(Payment) private paymentRepository: Repository<Payment>,
+  ) {}
+
+  async savePaymentRecord({
+    amount,
+    status,
+    transactionId,
+    bookingId,
+  }: PaymentRecord) {
+    const existingPaymentRecord =
+      await this.getPaymentRecordByBookingId(bookingId);
+
+    const newPaymentRecord = this.paymentRepository.create({
+      amount,
+      transactionId,
+      bookingId,
+    });
+
+    const paymentRecord = existingPaymentRecord
+      ? existingPaymentRecord
+      : newPaymentRecord;
+
+    paymentRecord.status = status;
+    await this.paymentRepository.save(paymentRecord);
+  }
+
+  async getPaymentRecordByBookingId(bookingId: string) {
+    const paymentRecord = this.paymentRepository.findOneBy({ bookingId });
```

Comment:
Will a new transaction be created if one is failed? 
As I understand from your code in payment.service, the answer is no, so you must add a unique constraint for bookingId to the payment table in DB. 
But if yes, there might be multiple transactions for a single booking, then you should add status checking too here.